temporal/error.go

```diff
@@ -174,6 +175,15 @@ func NewApplicationError(message, errType string, details ...interface{}) error
 	return internal.NewApplicationError(message, errType, false, nil, details...)
 }
 
+// NewApplicationErrorWithRetryDelay creates new instance of retryable *ApplicationErrorWithRetryRetryDelay with message,
+// type, retry delay and optional details.
+// Use ApplicationErrorWithRetryDelay to indicate that the activity needs to be restarted after retryDelay interval.
+// errType can be used to allow control from Workflow if error is retryable or not. Add the same type in
+// to RetryPolicy.NonRetryableErrorTypes to avoid retrying of particular error types.
+func NewApplicationErrorWithRetryDelay(message, errType string, retryDelay time.Duration, details ...interface{}) error {
+	return internal.NewApplicationErrorWithRetryDelay(message, errType, retryDelay, nil, details...)
```

Comment:
This logic is similar to on in NewApplicationError. Should not we receive the error from the user on NewApplicationErrorWithRetryDelay call?